### DISCUSSION TITLE
OpenAIP Integration Android

### DISCUSSION THREAD

**Discussion by SimoLP** [openAIP/openaip-public#326] [Q&A] [answered]
Hi,
I tryed with the map in C# but i didn't make it, so now i'm trying to integrate your map in an Android Studio (Java) app, but i don't know how to do that and how to fech the information of a point by clicking it and how to make routes to get (for example) the heading and the distance beetween two points.

Best Regards
Simo

**Comment by reskume** [comment, marked as answer]
Hi Simo,

unfortunately this is completely out of scope of openAIP. This question is related to the map client that you are using to implement that map feature inside your android app. Sorry, I can't help here and I suppose nobody can since this is a very specific question. But you definitely will find help in a forum related to the libraries that you are using.

Best,

Stephan

**Reply by SimoLP** [reply]
I understand.
And there is a way to do this in JS and with a webview integrate in andoid ?

**Comment by reskume** [comment]
I have no experience with Android or iOS - only with frontend/backend web development. I suppose you will find libs like mapboxgl for Android like this: https://docs.mapbox.com/android/maps/guides/

**Reply by SimoLP** [reply]
I know MapBox, but i was trying to find a free way

**Reply by reskume** [reply]
I'm out of ideas here - sorry. Depending on the expected popularity of the app, you may still be in the generous free tier of mapbox. We use it too and we get a lot of requests and until today,  we are still on the free tier. Best of luck!

**Reply by SimoLP** [reply]
Ok, thank you for your help,
Regards
